main.go

```diff
@@ -140,22 +143,36 @@ func mainE(ctx context.Context) error {
 		}
 	}
 
-	var azureMigratorFactory migration.MigratorFactory
+	var migratorFactory migration.MigratorFactory
 	{
-		azureMigratorFactory, err = migration.NewAzureMigratorFactory(migration.AzureMigrationConfig{
-			CtrlClient:    mgr.GetClient(),
-			Logger:        log,
-			TenantCluster: tenantCluster,
-		})
-		if err != nil {
-			return microerror.Mask(err)
+		if flags.Provider == "aws" {
+			migratorFactory, err = migration.NewAzureMigratorFactory(migration.AzureMigrationConfig{
```

Comment:
I guess you want to initialize AWSMigratoFactory here?

Reply:
yeah you are right , nice catch